Add age encryption

rokostik opened the pull request:
This PR implements everything needed to use age encryption.

There's also a small change with the readers in builtins_io.

This is just a draft preview, it doesn't work correctly yet.

refaktor reviewed (approved):
Great! Thank you for your work. If possible please also add an example of use at some point.

rokostik commented:
@refaktor I've added the tests and example in [2628ed0](https://github.com/refaktor/rye/pull/487/commits/2628ed0d9ee3fff1a8f76217e31dc6910fc69b38).

In [45cbfd6](https://github.com/refaktor/rye/pull/487/commits/45cbfd69ac3a671a575359b3d9818cee0ae25f05) I've added the missing data folder that includes the file.txt which is used by the io tests. Without it the docs wouldn't generate locally. The other two files committed were created by the tests.

This PR is now ready for review.

refaktor reviewed (approved):
Great changes and additions!

refaktor commented:
@rokostik You wrote in s3 example ... "how can we close the reader?".

In the longer term I intend to implement `defer` similar to what Go has but code would need to be in a block for consistency `defer { close r }`. 

Another or additional approach is something like `with/reader { do stuff with it } ; at exit of the block it's closed automatically, similar approach can be elegant for opening/closing files or sqlite databases for example.

In each case we should make generic "reader//close" function taht closes the reader so you could use it in code in a classical way before have those two.
